bridge/client/app/_guards/pending-changes.guard.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { CanDeactivate } from '@angular/router';
+import { Observable } from 'rxjs';
+
+export interface PendingChangesComponent {
+  canDeactivate: () => boolean | Observable<boolean>;
+  showNotification: ($event?: BeforeUnloadEvent) => void;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PendingChangesGuard implements CanDeactivate<PendingChangesComponent> {
+  canDeactivate(component: PendingChangesComponent): boolean | Observable<boolean> {
+    const hasPendingChanges = component.canDeactivate() !== true;
```

Comment:
If it's an `Observable` it's always `false`.

Reply:
But if no changes are applied to the Form it is set to `true`.

Reply:
Yes, because the value is `boolean` then, but on change it is set to `Observable<boolean>` and `of(true) !== true` and `of(false) !== true`. It would be the same if `false` is set instead of assigning an observable to it

Reply:
Yes, but here the `hasPendingChanges` is just used to show the Notification. But the `Observable<boolean>` is also returned by the Guard, so the Route change can be triggered after e.g. saving the form or declined by the user. Without `Observable` this will not work.